### PR TITLE
Adds tests for ARM clientauthoriser

### PR DESCRIPTION
**Pull request by m1kola**

### CODE REVIEW

File: pkg/util/clientauthorizer/arm_test.go [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+	"net/http"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+func TestARM(t *testing.T) {
+	t.Run("IsReady", func(t *testing.T) {
```
Comment (jim-minter) [inline pkg/util/clientauthorizer/arm_test.go]: make these separate top level level `func TestFoo`s please

File: pkg/util/clientauthorizer/arm_test.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+		log := logrus.NewEntry(logrus.StandardLogger())
+		fakeNow, _ := time.Parse(time.RFC3339, "2020-01-20T00:00:00Z")
+		fakeNowFunc := func() time.Time { return fakeNow }
+
+		tests := []struct {
+			name string
+			do   func(req *http.Request) (*http.Response, error)
+			want bool
```
Comment (jim-minter) [inline pkg/util/clientauthorizer/arm_test.go]: I think this would be better to be reversed, i.e. wantErr string

**Review by jim-minter — CHANGES_REQUESTED** [review]
looks good, just nits

**Comment by m1kola** [comment]
@jim-minter PTAL, I updated and rebased on top of the master.

**Comment by jim-minter** [comment]
merged, thanks

**Pull request closed without merging**